### DISCUSSION TITLE
question about model.optimize()

### DISCUSSION THREAD

**Discussion by amarsh26** [opencobra/cobrapy#1382] [General]
Hello,

I observe the following when I use iSB619 from the BiGG data base. 

This code 
```
isb_table=iSB619.optimize()

print(isb_table['DAPDC'])

```
provides the following output: 
2.046751276255936

However, this code
```
isb_table=iSB619_model.slim_optimize()
print(isb_table['DAPDC'])
```

produces the following output. 

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [66], in <cell line: 1>()
----> 1 print(isb_table['DAPDC'])

TypeError: 'float' object is not subscriptable

Further, both methods work for other BiGG models such as iML1515 and iJO1366. I was wondering the origin of the TypeError in the second case for iSB619.

Thank you!

**Comment by Midnighter** [comment]
Have a look at the documentation for the [`optimize`](https://cobrapy.readthedocs.io/en/latest/autoapi/cobra/core/model/index.html#cobra.core.model.Model.optimize) and the [`slim_optimize`](https://cobrapy.readthedocs.io/en/latest/autoapi/cobra/core/model/index.html#cobra.core.model.Model.slim_optimize) methods. The former returns a solution object which supports accessing reaction fluxes in the way that you did. The latter only returns the objective value if it exists. You cannot perform the access on a float as you are trying in your second example, hence, the type error.

**Comment by amarsh26** [comment]
Thanks so much for your response. 

I just wanted to follow up in why it would work for iJO1366	(0.9823718) and iML1515 (0.8769972) both of which are floats. 

Thanks again!
